state-chain/pallets/cf-flip/src/on_charge_transaction.rs

```diff
@@ -67,6 +88,10 @@ impl<T: TxConfig + FlipConfig + Config> OnChargeTransaction<T> for FlipTransacti
 				surplus.peek()
 			};
 
+			let to_burn =
+				T::TransactionFeeScaler::get_fee_multiplier(escrow.scalable_fee_call_info)
+					.saturating_mul_int(to_burn);
```

Comment:
Better than just returning a multiple would be to have the scaler implement some scaling function and separate the state access from the scaling implementation:

```suggestion
			// (See my other comment re. CallCounter<T>)
			let scale_factor = CallCounter<T: Config>::mutate(|c| { c += 1; c });
			let to_burn =
				T::TransactionFeeScaler::scale_fee(to_burn, scale_factor);
```